### PR TITLE
Replace Deprecated .getBuildDir with .getLayout().getBuildDirectory() and make Lazy

### PR DESCRIPTION
**Pull request by JT122406**
image

### CODE REVIEW

**Review by modmuss50 — APPROVED** [review]

**Comment by Juuxel** [comment]
See also #970, which would be simple to do together with this change.

**Comment by JT122406** [comment]
Do I need to do something to fix the failed tests or?

**Comment by modmuss50** [comment]
The failing tests are either random network failuires, or known flakey tests, so dont worry about them. All of the important ones have passed.

**Comment by JT122406** [comment]
Anything else need to be done for this to get merged?

**Comment by modmuss50** [comment]
> Anything else need to be done for this to get merged?

Are you able to make it lazy as suggested in #970 ?

**Comment by JT122406** [comment]
> > Anything else need to be done for this to get merged?
> 
> Are you able to make it lazy as suggested in #970 ?

I can give it a shot,
Wasn't sure if you wanted that in this or separate pr

**Comment by modmuss50** [comment]
> > > Anything else need to be done for this to get merged?
> > 
> > 
> > Are you able to make it lazy as suggested in #970 ?
> 
> I can give it a shot, Wasn't sure if you wanted that in this or separate pr

Let me know if you have any issues, I think it makes sense to do as part of this PR as it changes the same line of code.

File: src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java [hunk 1]
```diff
@@ -92,7 +92,7 @@ public void run() {
 		// Configure the default jar task
 		getTasks().named(JavaPlugin.JAR_TASK_NAME, AbstractArchiveTask.class).configure(task -> {
 			task.getArchiveClassifier().convention("dev");
-			task.getDestinationDirectory().set(new File(getProject().getBuildDir(), "devlibs"));
+			task.getDestinationDirectory().set(new File(getProject().getLayout().getBuildDirectory().getAsFile().get(), "devlibs"));
```
Comment (modmuss50) [inline src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java]: You can make these lazy like so, removes the need to create it as a `File` as well.
```suggestion
			task.getDestinationDirectory().set(getProject().getLayout().getBuildDirectory().map(directory -> directory.dir("devlibs")));
```

**Comment by modmuss50** [comment]
Can you update this PR to make these use a lazy provider? I dont seem to be able to push the change to your branch. Should be good to merge once done.

**Comment by JT122406** [comment]
> Can you update this PR to make these use a lazy provider? I dont seem to be able to push the change to your branch. Should be good to merge once done.

You should be able to push to my branch

File: src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java [hunk 2]
```diff
@@ -139,7 +139,7 @@ private void trySetupSourceRemapping() {
 			}
 
 			sourcesJarTask.getArchiveClassifier().convention("dev-sources");
-			sourcesJarTask.getDestinationDirectory().set(new File(getProject().getBuildDir(), "devlibs"));
+			sourcesJarTask.getDestinationDirectory().set(new File(getProject().getLayout().getBuildDirectory().getAsFile().get(), "devlibs"));
```
Comment (modmuss50) [inline src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java]: Sorry, I wasnt clear, this one also needs updating.

**Comment by modmuss50** [comment]
> > Can you update this PR to make these use a lazy provider? I dont seem to be able to push the change to your branch. Should be good to merge once done.
> 
> You should be able to push to my branch

I tried, for some reason it wouldnt allow me to. Maybe due to the fact its the default branch in your repo I'm not sure.

**Comment by JT122406** [comment]
Should be good now

**Comment by JT122406** [comment]
I'm assuming we don't want lazy in LoomFilesProjectImpl.java

File: src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java (hunk 2 above)
Comment (JT122406) [reply src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java]: Nope you were clear just had to fix the repo on my side since it was out of date

**Comment by modmuss50** [comment]
> I'm assuming we don't want lazy in LoomFilesProjectImpl.java

No, this looks good now. Thanks. I hope it builds, as I think I had to remove the File import as its no longer required.

**Comment by JT122406** [comment]
> > I'm assuming we don't want lazy in LoomFilesProjectImpl.java
> 
> No, this looks good now. Thanks. I hope it builds, as I think I had to remove the File import as its no longer required.

Just handled that

**Comment by JT122406** [comment]
Sorry about the wait with this pr kinda forgot about it with everything going on with finals, Thanks for reminding me @modmuss50

**Comment by modmuss50** [comment]
No worries at all, thanks a lot. I've also been busy with other things and have only just got around to looking at loom things again 👍

**Pull request merged**